File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/ExceptionalService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.ExceptionalDTO;

import java.util.List;

public interface ExceptionalService {

    void addExceptional(ExceptionalDTO exceptional);

    void updateExceptional(ExceptionalDTO exceptional);

    void deleteExceptional(Long id);

    List<ExceptionalDTO> getAllExceptionals();

    List<ExceptionalDTO> findExceptionalsByCalendarId(Long id);
}

File: CartagoFrontV2-master/src/app/services/referenciel/exceptional.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Exceptional } from '../../models/exceptional';

@Injectable({
  providedIn: 'root',
})
export class ExceptionalService {
  constructor(private http: HttpClient) {}

  getAllExceptionals(): Observable<Exceptional[]> {
    return this.http.get<Exceptional[]>('http://localhost:8080/Exceptionals');
  }

  addExceptional(exceptional: Exceptional): Observable<Exceptional> {
    return this.http.post<Exceptional>(
      'http://localhost:8080/Exceptionals/',
      exceptional
    );
  }

  updateExceptional(exceptional: Exceptional): Observable<Exceptional> {
    return this.http.put<Exceptional>(
      'http://localhost:8080/Exceptionals/',
      exceptional
    );
  }

  deleteExceptional(id: number): Observable<Exceptional> {
    return this.http.delete<Exceptional>(
      'http://localhost:8080/Exceptionals/' + id
    );
  }
}
